contract-message-test/src/contract_message_test.sw

```diff
@@ -0,0 +1,70 @@
+contract;
+
+dep contract_message_test_abi;
+dep utils;
+
+use contract_message_receiver::MessageReceiver;
+use contract_message_test_abi::VerifyMessageData;
+use utils::{input_message_data, input_message_data_length};
+use std::address::Address;
+use std::contract_id::ContractId;
+use std::constants::ZERO_B256;
+
+storage {
+    counter: u64 = 0,
+    data1: ContractId = ~ContractId::from(ZERO_B256),
```

Comment:
Logically that makes sense, especially for more complex contracts! I think I like this simpler setup for just a test though. I'm also wondering how much this would effect contract storage usage. Is storage stored per word? Does this mean making something "Optional" add an additional storage slot?

Reply:
I don't recall exactly how storage is handled so I do not want to misinform.
An option is an enum which contains a tag (8 bits) and then there's space for the largest type so an option would take up as much room as the space for the tag + space for the largest type in the enum. How that gets handled actually in storage is beyond my recall however there has been some talk about enums in storage potentially being changed to use 1 slot. @nfurfaro and @mohammadfawaz should be able to elaborate since I recall seeing their comments on some issues.

Reply:
Yes the tag of the enum will always take up 1 storage slots. Nothing else goes in that slot. The variants themselves are sent to a different slot (or set of slots) depending on how large the largest variant is.

That being said, this has been proposed: https://github.com/FuelLabs/fuel-specs/issues/395